DiscordBot/Commands/EventCommands.cs

```diff
@@ -19,34 +20,109 @@ internal class EventCommands
         };
 
         [Command("event")]
+        [Description("Initiates the wizard for event-related actions")]
         public async Task Event(CommandContext context)
         {
+            var interactivity = context.Client.GetInteractivityModule();
+
             await context.RespondAsync(
                 $"{context.Member.Mention} - choose one of the actions below or answer ``stop`` to cancel. (time out in 30s)\n" +
                 "``create`` - create new event.\n" +
                 "``rm`` - delete event.\n" +
                 "``edit`` - edit event.\n"
             );
 
-            var interactivity = context.Client.GetInteractivityModule();
-            var response = await interactivity.WaitForMessageAsync(
-                message =>
-                    IsValidResponse(message.Content),
-                TimeSpan.FromSeconds(30));
+            var eventOperation = await GetUserResponse(context, interactivity, EventOperations);
+            if (eventOperation == null)
+            {
+                return;
+            }
+
+            switch (eventOperation)
+            {
+                case "create":
+                    await CreateEvent(context, interactivity);
+                    break;
+                case "rm":
+                    await context.RespondAsync("Not implemented yet!");
+                    break;
+                case "edit":
+                    await context.RespondAsync("Not implemented yet!");
+                    break;
+            }
+        }
 
-            if (response == null)
+        public async Task CreateEvent(CommandContext context, InteractivityModule interactivity)
+        {
+            await context.RespondAsync($"{context.Member.Mention} - what is the name of the event?");
+            var eventName = await GetUserResponse(context, interactivity);
+            if (eventName == null)
             {
-                await context.RespondAsync($"{context.Member.Mention} - timed out");
+                return;
             }
-            else
+            
+            await context.RespondAsync($"{context.Member.Mention} - give an event description.");
+            var eventDescription = await GetUserResponse(context, interactivity);
+            if (eventDescription == null)
             {
-                await context.RespondAsync(response.Message.Content);
+                return;
             }
+
+            DateTimeOffset eventTime;
+
+            while (true)
```

Comment:
Why is just this bit in a while loop? 🙂

Reply:
That section is currently being looped until a response that can be parsed as a time is given

Reply:
I'm not sure a loop is necessary, I think just a descriptive error if the time can't be parsed should be enough 🙂